packages/terra-pills/src/Pills.jsx

```diff
@@ -0,0 +1,383 @@
+      PillsUtils.setPillsTabIndex(pills, '-1');
+
+      // Determine the first focusable pill
+      if (firstFocusableNode !== -1) {
+        focusNode.current = firstFocusableNode;
+        currentPill.current = pills[focusNode.current].id;
+        setTabIndex('0');
+      } else if (isSingleLine && rollUpPill) {
+        currentPill.current = rollUpPill.getAttribute('id');
+        setTabIndex('0');
+      }
+    } else if (isSingleLine && rollUpPill && pills.length === 0) { // if the first pill is rollUp pill, set rollUp pill tabindex 0
+      currentPill.current = rollUpPill.getAttribute('id');
+      setTabIndex('0');
+      focusNode.current = 0;
+    }
+  }, [isSingleLine, updatedCount]);
+
+  useEffect(() => {
+    const pills = [...pillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = pillsRef.current.querySelector('[data-terra-rollup-pill]');
+    const nextFocusableNode = PillsUtils.getNextFocusableNode(pills, focusNode.current);
+    const prevFocusableNode = PillsUtils.getPreviousFocusableNode(pills, focusNode.current);
+    const startIndex = PillsUtils.handleRollUp(pillsRef);
+
+    if (isPillDeleted.current && React.Children.count(children) > 0) {
+      if (nextFocusableNode === -1 && prevFocusableNode === -1 && rollUpPill === null) {
+        focusPillsContainer();
+        return;
+      }
+      // if there is a roll Up pill, set tabindex to 0
+      if (rollUpPill) {
+        PillsUtils.setRollUpPillTabIndex(rollUpPill, '-1');
+      }
+
+      if (pills.length > 0) {
+        PillsUtils.setPillsTabIndex(pills, '-1');
+      }
+
+      if (rollUpPill && startIndex >= 1) {
+        currentPill.current = pills[focusNode.current].id;
+      } else if (rollUpPill) {
+        currentPill.current = rollUpPill.getAttribute('id');
+      } else if (prevFocusableNode >= 0) {
+        focusNode.current = prevFocusableNode;
+        currentPill.current = pills[focusNode.current].id;
+      } else {
+        focusNode.current = nextFocusableNode;
+        currentPill.current = pills[focusNode.current].id;
+      }
+      setTabIndex('0');
+      focusCurrentNode();
+      isPillDeleted.current = false;
+    }
+  }, [children, isPillDeleted]);
+
+  useEffect(() => {
+    const pills = [...pillsRef.current.querySelectorAll('[data-terra-pill]')];
+
+    // To focus the immediate focusable pill after the rollUp pill is selected/removed
+    if (isRollUpRemoved.current) {
+      if (pills.length === React.Children.count(children)) {
+        setTabIndex('-1');
+        focusNode.current += 1;
+        currentPill.current = pills[focusNode.current].id;
+        setTabIndex('0');
+        focusCurrentNode();
+        isRollUpRemoved.current = false;
+      }
+    }
+  }, [children, updatedCount]);
+
+  const focusNextNode = (pills, rollUpPill) => {
+    const allPills = React.Children.toArray(children);
+    const nextFocusableNode = PillsUtils.getNextFocusableNode(pills, focusNode.current + 1);
```

Comment:
```suggestion
    const nextFocusableNode = PillsUtils.getNextFocusableNode(pills, focusNode.current);
```

Reply:
But in case when there is one selectable pill and a rollup pill. if the focus is on first selectable pill. then current 'getNextFocusableNode' returns  '-1 ' if we pass 'focusNode.current + 1' here. it should return 0 if we are using pills index or 1 if we are going with the pills count.

Reply:
If there is just one focusable pill and a rollup pill, clicking on "right-arrow" should focus the rollup pill.
In this case you are right,  "getNextFocusableNode' returns '-1'.  But since there is just one pill, the block between L213 - L 218 is hit and is returned. The negative value of '-1' is neglected.

Reply:
How about we update the function to something like this:
```jsx
  const focusNextNode = (pills, rollUpPill) => {
    const allPills = React.Children.toArray(children);
    const nextFocusableNode = PillsUtils.getNextFocusableNode(pills, focusNode.current + 1);

    // focus the next pill
    if (focusNode.current + 1 < pills.length || focusNode.current + 1 === pills.length) {
      setTabIndex('-1');
      // if there no focusable pills or the next pill is rollUpPill, focus the rollUpPill
      if ((nextFocusableNode === focusNode.current || focusNode.current + 1 === pills.length) && rollUpPill) {
        currentPill.current = rollUpPill.getAttribute('id');
      } else if (nextFocusableNode !== -1) {
        focusNode.current = nextFocusableNode;
        currentPill.current = allPills[focusNode.current].props.id;
      }
      setTabIndex('0');
      focusCurrentNode();
    }
  };
```